File: enemy.py
'''A non-protagonist character.'''

import character
import pygame
import random

pygame.init()

# Image cache.
IMAGE_CACHE = {
    "steven": pygame.image.load("art/male-head.png"),
    "sierra": pygame.image.load("art/female-head.png"),
    "rabbit": pygame.image.load("art/rabbitside.png")
    }

class Creep(character.Character):
    '''A creep that is not the protagonist.'''
    def __init__(self, creep_type, x, y):
        super().__init__(x, y)
        self.type = creep_type

def generate_creep(x, y):
    creep_type = random.choice(list(IMAGE_CACHE.keys()))
    return Creep(creep_type, x, y)

File: js/app/enemy.js
// Generates enemy characters.

define(['./character'], function(character) {
  "use strict";

  const CREEPS = ["steven", "sierra", "rabbit", "cat"];
  function generate_creep(x, y) {
    const creep_type = CREEPS[Math.floor(CREEPS.length * Math.random())];
    return character.Character.create(creep_type, [x, y]);
  }

  return {generate_creep};
});
